src/peerbook/libp2p_peerbook.erl

```diff
@@ -200,12 +200,43 @@ random(Peerbook, Exclude) ->
 random(Peerbook, Exclude, Pred) ->
     random(Peerbook, Exclude, Pred, 15).
 
+
+iterator_move_filtered(KeyFun, Iterator, Action) ->
+    case rocksdb:iterator_move(Iterator, Action) of
```

Comment:
Also fwiw, if the iterator is built with seek option. This could loop back to the beginning and when it does that the guard pattern changes. It flips the key and value. Atleast from what I’ve experienced.

Reply:
that seems bizarre?

Reply:
I tried to look to see why this happens in the source for rocksDB but it doesn't jump out to me. I just encountered this while working on my ETL optimizations that I had done not to long ago.

Reply:
The docs have some caveats about undefined behavior around using seek iterator when the database or column family was opened with a prefix option, I don't think we are using the prefix option for peerbook though.